### PR TITLE
Add tests for join methods

### PR DESCRIPTION
**Pull request by willsoto**
#1431

### CODE REVIEW

File: tests/ts/query-builder-api/join-methods.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    .where('pr:pt.species', 'dog');
+
+  await Person.query().innerJoinRelation('pets');
+  await Person.query().outerJoinRelation('pets');
+  await Person.query().leftJoinRelation('pets');
+  await Person.query().leftOuterJoinRelation('pets');
+  await Person.query().rightJoinRelation('pets');
+  await Person.query().rightOuterJoinRelation('pets');
+  await Person.query().fullOuterJoinRelation('pets');
+  await Person.query().join(raw('pets'));
```
Comment (willsoto) [inline tests/ts/query-builder-api/join-methods.ts]: Typings don't currently allow `.join('pets')` - not sure if intentional or oversight. Happy to fix if an oversight.

Comment (koskimas) [reply tests/ts/query-builder-api/join-methods.ts]: They are not supposed to support that. `*joinRelation` methods take relation names. Other join methods simply call the knex equivalents and know nothing about relations.

**Comment by willsoto** [comment]
This should be ok to merge, that was the only thing I had a question about.

**Comment by koskimas** [comment]
Thanks again!

**Pull request merged**